renderEngine/Command/Source/BaseClasses/SqlMember.js

```diff
@@ -0,0 +1,93 @@
+import InMemoryMember from "./InMemoryMember.js";
+import JsonSource from "../../../Source/JsonSource.js";
+import IContext from "../../../Context/IContext.js";
+import alasql from "alasql";
+import ContextBase from "../../../Context/ContextBase.js";
+export default class SqlMember extends InMemoryMember {
+  /**
+   * @param {object} memberIL
+   */
+  constructor(memberIL) {
+    super(memberIL);
+  }
+  /**
+   *
+   * @param {IContext} context
+   * @returns {JsonSource}
+   */
+  async _parseDataAsync(context) {
+    try {
+      let db = new alasql.Database();
+      const content = await this.rawContent.getValueAsync(context);
+      const dataMemberNames = this.findDataMembersInContent(content);
+      const getSourcePromises = dataMemberNames.map((dataMemberName) =>
+        context.waitToGetSourceAsync(dataMemberName)
+      );
+      const sources = await Promise.all(getSourcePromises);
+      const insertPromises = sources.map(async (source, index) => {
+        const dataArray = source.data;
+        const dataMemberName = dataMemberNames[index];
+        const createTableSql = `CREATE TABLE [${dataMemberName}] (${Object.keys(
+          dataArray[0]
+        )
+          .map((key) => `${key} STRING`)
```

Comment:
we have columns property in [IDataSource  ](https://github.com/Manzoomeh/BasisCore.Server.Node/blob/d6ecfc32c0fe024ce3ead6e97b9e031da6b4a74a/renderEngine/Source/IDataSource.js#L9C6-L9C16)
```javascript
/**@type {Array<string>} */
  get columns() {
    return this.data?.length > 0 ? Object.keys(this.data[0]) : [];
  }
```

use it please.

Reply:
also all column is not string. use `STRING` for all maybe cause error. please add test for case like `join over number` column

Reply:
i added that ; thanks.